### PR TITLE
Feature/testing

### PR DESCRIPTION
**Pull request by patriciachrysy**
Hello Dear reviewer,
thanks for taking time to review my code, feedbacks are welcomed. Here is what we did as steps to complete this project:

- Addes rspec gem for testing
- Added test cases for Book, Classroom, Person, Rental, Teacher and Student classes
Happy review :-)

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  it 'should add two rentals for the book' do
+    book.add_rental(rental_date_one, unknownPerson)
+    book.add_rental(rental_date_two, person)
+    expect(book.rentals.length).to eq(2)
+  end
+
+  it 'should find Cinderela book among the list of books' do
+    expect(Book.all[0].title).to eq('Cinderela')
+  end
+end
```
Comment (ridwanediallo) [inline spec/book_spec.rb:36]: - [ ] Well done so far in testing the book class 👏. However, as requested in the project requirement, all methods inside the class should be tested. So ensure as well that the `add_rental` and the `self.all` are tested. 👌

### Requirement :point_down: 
Screenshot from 2023-11-16 14-31-50

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  it 'should initialize the unknown_person with Unknown name' do
+    expect(unknown_person.name).to eq('Unkown')
+  end
+
+  it 'should state Sally cannot use the services but unkown can' do
+    expect(person.can_use_services?).to eq(false)
+    expect(unknown_person.can_use_services?).to eq(true)
+  end
```
Comment (ridwanediallo) [inline spec/person_spec.rb:24]: - [x] Same for the `Person` class, please ensure that the `add_rental` , `self.all` , `correct_name`and `of_age?` methods are tested too. 👌

### Requirement :point_down: 
Screenshot from 2023-11-16 14-31-50

File: spec/rental_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    expect(rental.person).to eq(person)
+    expect(rental.book).to eq(book)
+  end
+  it 'should add this rental to Sallys' do
+    expect(person.rentals).to include(rental)
+  end
+  it 'should add this rental to Cinderella books' do
+    expect(book.rentals).to include(rental)
+  end
+end
```
Comment (ridwanediallo) [inline spec/rental_spec.rb]: - [x] Same for the `Rental` class; please ensure that the `self.all` method is tested too. 👌

### Requirement :point_down: 
Screenshot from 2023-11-16 14-31-50

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi :wave team, 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Awesome! All the tests so far are successfull. ✔


## Required Changes ♻️

- [ ] According to the requirments the methods inside the classes created while implementing the Decorator design pattern should as well be tested. So kindly make  sure that the `correct_name` inside all the following classes is tested.🤗
1. The `Nameable` class.
2. The `BaseDecorator` class. 
3. The `TrimmerDecorator` class.
4. The `CapitalizeDecorator` class.




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb (hunk 1 above)
Comment (patriciachrysy) [reply spec/book_spec.rb:36]: Please check the file well all the methods are tested

Comment (ridwanediallo) [reply spec/book_spec.rb:36]: Hello, @patriciachrysy , my apologies. Please ignore this comment and check for the remaining ones.

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @patriciachrysy, 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡
 
✅ Descriptive PR 👍 
✅ Implement requested changes 👍 


## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**